r/R/table.R

```diff
@@ -339,7 +339,10 @@ as_arrow_table.Dataset <- function(x, ...) {
 #' @rdname as_arrow_table
 #' @export
 as_arrow_table.arrow_dplyr_query <- function(x, ...) {
-  out <- as_arrow_table(as_record_batch_reader(x))
+  reader <- as_record_batch_reader(x)
+  on.exit(reader$.unsafe_delete())
```

Comment:
Similarly, should we do this everywhere a record batch reader is created an consumed within a function? For example:

https://github.com/apache/arrow/blob/37a79659f82a7dbe0a55858e59fa5eaa380a266f/r/R/dplyr.R#L297

https://github.com/apache/arrow/blob/37a79659f82a7dbe0a55858e59fa5eaa380a266f/r/R/python.R#L111